src/components/subcomponents/upload.js

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from "react";
```

Comment:
Honestly, this file is huge and has a lot going on. I think we should consider breaking this up with redux.

Pseudo code for a fileUpload thunk
1. fire UPLOAD_STARTED action
2. get upload url
3. start upload
4. track upload progress and fire UPLOAD_PROGRESS actions (https://gist.github.com/virolea/e1af9359fe071f24de3da3500ff0f429)
5. fire UPLOAD_FINISHED action
6. if any errors, fire UPLOAD_ERROR action (we definitely need error handling for file uploads, things go wrong with big files).